src/test/java/teammates/test/cases/search/FeedbackResponseCommentSearchTest.java

```diff
@@ -0,0 +1,97 @@
+package teammates.test.cases.search;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.FeedbackResponseCommentSearchResultBundle;
+import teammates.common.datatransfer.attributes.FeedbackResponseCommentAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.storage.api.FeedbackResponseCommentsDb;
+
+/**
+ * SUT: {@link FeedbackResponseCommentsDb},
+ *      {@link teammates.storage.search.FeedbackResponseCommentDocument},
+ *      {@link teammates.storage.search.FeedbackResponseCommentSearchQuery}.
+ */
+public class FeedbackResponseCommentSearchTest extends BaseSearchTest {
+
+    @Test
+    public void allTests() {
+        FeedbackResponseCommentsDb commentsDb = new FeedbackResponseCommentsDb();
+
+        FeedbackResponseCommentAttributes frc1I1Q1S1 = dataBundle.feedbackResponseComments
+                .get("comment1FromT1C1ToR1Q1S1C1");
+        FeedbackResponseCommentAttributes frc1I1Q2S1 = dataBundle.feedbackResponseComments
+                .get("comment1FromT1C1ToR1Q2S1C1");
+        ArrayList<InstructorAttributes> instructors = new ArrayList<InstructorAttributes>();
+        instructors.add(dataBundle.instructors.get("instructor1OfCourse1"));
+
+        ______TS("success: search for comments in whole system; query string does not match any comment");
+
+        FeedbackResponseCommentSearchResultBundle bundle = commentsDb.search("non-existent", instructors);
+        assertEquals(0, bundle.numberOfResults);
+        assertTrue(bundle.comments.isEmpty());
+
+        ______TS("success: search for comments in whole system; query string matches single comment");
+
+        bundle = commentsDb.search("\"Instructor 1 comment to student 1 self feedback Question 2\"", instructors);
+        verifySearchResults(bundle, frc1I1Q2S1);
+
+        ______TS("success: search for comments in instructor's course; query string matches some comments");
+
+        bundle = commentsDb.search("\"self feedback\"", instructors);
+        verifySearchResults(bundle, frc1I1Q1S1, frc1I1Q2S1);
+
+        ______TS("success: search for comments in instructor's course; query string is case in-sensitive");
+
+        bundle = commentsDb.search("\"Instructor 1 COMMENT to student 1 self feedback Question 2\"", instructors);
+        verifySearchResults(bundle, frc1I1Q2S1);
+
+        ______TS("success: search for comments searchable by feedbackSessionName");
```

Comment:
This `TS` is phrased differently from the next two. Are they doing similar things? In that case, phrase them the same way.